app/scripts/controllers/detect-tokens.js

```diff
@@ -72,16 +74,24 @@ export default class DetectTokensController extends PollingControllerOnly {
     this.chainId = this.getChainIdFromNetworkStore();
     this._trackMetaMetricsEvent = trackMetaMetricsEvent;
 
-    preferences?.store.subscribe(({ selectedAddress, useTokenDetection }) => {
-      if (
-        this.selectedAddress !== selectedAddress ||
-        this.useTokenDetection !== useTokenDetection
-      ) {
-        this.selectedAddress = selectedAddress;
-        this.useTokenDetection = useTokenDetection;
-        this.restartTokenDetection({ selectedAddress });
-      }
-    });
+    messenger.subscribe(
+      'AccountsController:selectedAccountChange',
+      (account) => {
+        const useTokenDetection =
+          this.preferences?.store.getState().useTokenDetection;
```

Comment:
What if the `useTokenDetection` preference changes? It looks like previously token detection would get restarted and now that would no longer happen.

Reply:
Ran a quick test with 801eeab commit and observed that token detection performs similarly in this branch to what I see on 11.5.1. When it is first enabled, it takes a few moments for tokens to be detected.  However, after that initial delay the UI promptly reflects any subsequent changes to the setting under security and privacy.

Reply:
In the following lines we still have the same logic @mcmire or am I missing something here?

Reply:
Solved in [1c15365](https://github.com/MetaMask/metamask-extension/pull/21437/commits/1c15365b7c1e857aa01c7b88c8700b1353eb3fb2) and [3b37cfa](https://github.com/MetaMask/metamask-extension/pull/21437/commits/3b37cfabc666d3d4f47e3863351d89893e72a919)

Reply:
Gotcha thanks!